assets/terraform-modules/packet/flatcar-linux/kubernetes/workers/cl/worker.yaml.tmpl

```diff
@@ -421,6 +434,35 @@ storage:
           done
           echo "$record.$zone is available on all nameservers"
           exit 0
+    - path: /opt/wait-for-bgp
+      filesystem: root
+      mode: 0544
+      contents:
+        inline: |
+          #!/bin/bash
+          set -o pipefail
+          max_attempts=3600
+          echo "Polling metadata service for BGP metadata"
+          counter=0
+          while [[ $counter -lt $max_attempts ]]; do
+              out=$(curl -s -f --connect-timeout 5 \
+                  https://metadata.packet.net/metadata | jq -r .bgp_neighbors[0].peer_ips[0])
+              ret=$?
+              if [[ $ret -ne 0 ]]; then
+                  echo "Non-zero exit code: $ret"
+              elif [[ "$out" = "" ]]; then
+                  echo "Empty response"
+              elif [[ "$out" = "null" ]]; then
+                  echo "Null response"
+              else
+                  echo "BGP metadata is available!"
```

Comment:
Wouldn't it be simpler to write the metadata info to an env file, so that the kubelet just uses that (like Enviromnent=<file created>) instead of repeating the curl?

Furthermore, the following curl can return an error, for example, and in that case we are not handling it correctly

Reply:
I guess it make sense, but we need to write it to some tmpfs, so it gets removed on boot, right?

Reply:
No, if the file is removed the kubelet service won't work after node restart. We want to preserve that file.

Reply:
Yeah, but on next boot it will be freshly recreated. Otherwise kubelet will start without waiting for file, so we may get stale data.

Reply:
>Otherwise kubelet will start without waiting for file, so we may get stale data.

I don't think there is ever a case where a node's BGP address changes.

Reply:
Yes, I've considered that but decided to go with something simpler since the whole thing is a workaround which should hopefully be gone soon. It *is* safer though to go with the file approach. I'll refactor.

Reply:
Done.